R/ancombc_prep.R

```diff
@@ -30,7 +30,7 @@ tse_construct = function(data, assay_name, tax_level, phyloseq) {
 
         # Check if agglomeration should be performed
         if (is.null(tax_level)) {
-            tax_levels = mia::taxonomyRanks(tse)
+            tax_levels = union(mia::taxonomyRanks(tse), rownames(SummarizedExperiment::rowData(tse)))
```

Comment:
Hmm I think this is mistaken. The rownames refer to individual taxa (rownames of rowData), not the high-level categories of those taxa (colnames of rowData). Moreover, we already have functionality to agglomerate TreeSE objects based on any rowData field, through the function `mia::mergeFeatures()` and for taxonomic ranks specifically via `mia::mergeFeaturesByRank()`. The internal function `mia::.merge_features` was intended to check whether the given field is in taxonomyRanks or not, and then choose the merging function based on that. That was the entire motivation for the recent PR work in mia. Therefore there is no need to take union of different field names here but instead we should use the background work that was already laid out in mia to automatically handle these things, if possible.

Reply:
I think that `mia::mergeFeatures()` automatically checks whether the field is in taxonomy ranks or not and then calls the internal function based on that.

Reply:
ACK. mia::mergeFeaturesByRank was not exported. There is a PR for that now. Similarly mia::.merge_features is not exported.  mia::mergeFeatures() deprecated mia::mergeRows.
The union to give tax_levels is to display in the message the possible tax_level params  when it is NULL.

Reply:
The `.merge_features` is internal to mia and should not be imported. But I think it can also be called if necessary, like with `mia:::.merge_features` (/w three double colons). To check. 

Let's first merge the relevant PRs in mia and solve this based on those. 

We can discuss in Slack or in mia issues whether we should have exported function that combines these two, or do we want to always separate them and create different combinations for different situations and packages like here now?